src/guiVolumeChange.cpp

```diff
@@ -124,41 +133,64 @@ void GUIVolumeChange::drawMenu()
 	gui::IGUIElement::draw();
 }
 
+bool GUIVolumeChange::acceptInput()
+{
+	gui::IGUIElement *e = getElementFromId(ID_soundMuteButton);
+	if (e != NULL && e->getType() == gui::EGUIET_CHECK_BOX)
+		g_settings->setBool("mute_sound", ((gui::IGUICheckBox*)e)->isChecked());
+
+	return true;
+}
+
 bool GUIVolumeChange::OnEvent(const SEvent& event)
 {
 	if (event.EventType == EET_KEY_INPUT_EVENT) {
 		if (event.KeyInput.Key == KEY_ESCAPE && event.KeyInput.PressedDown) {
+			acceptInput();
 			quitMenu();
 			return true;
 		}
 
 		if (event.KeyInput.Key == KEY_RETURN && event.KeyInput.PressedDown) {
+			acceptInput();
 			quitMenu();
 			return true;
 		}
-	}
-
-	if (event.GUIEvent.EventType == gui::EGET_BUTTON_CLICKED) {
-		if (event.GUIEvent.Caller->getID() == ID_soundExitButton) {
-			quitMenu();
-			return true;
+	} else if (event.EventType == EET_GUI_EVENT) {
+		if (event.GUIEvent.EventType == gui::EGET_ELEMENT_FOCUS_LOST
```

Comment:
You could simply check for `EGET_CHECKBOX_CHANGED` and update the `mute_sound` setting inside.

Reply:
Okay, makes sense to apply it immediately when clicking the checkbox